### PR TITLE
Dependency updates 20190718

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

Most of these are standard but but the analytics/okhttp change is vital to support our remaining users on Android 4.x - I inadvertently snuck in an update to okhttp 4.x which is API19+ only - this fixes it but dropping okhttp back to 3.x-current

## Fixes
@milyukhin noticed it and mentioned this in #5021 (thank you Max!)

## How Has This Been Tested?

test suite in analytics package passed, and test suite here passes as well, and it's a major version of okhttp that was in use before

## Other notes

@timrae note that you seem to have some local changes (related to the 2.9 betas?) that are not reflected in upstream/master? Or maybe my local state is a little off - but I don't seem to have commits that correspond to the tags

### CODE REVIEW

**Comment by timrae** [comment]
Yes, the branches will in general diverge from this point onwards. PR's will be merged into master, and then cherry-picked into the release branch. 2.9 version bump commits and localisation updates will not be in master going forwards.

**Comment by timrae** [comment]
It looks like 4 of the test matrix builds failed

**Comment by mikehardy** [comment]
Oh of course, branch now. Been a while! I'll make sure to check master and 2.9 then and I'll check the CI fail of course. Thanks

**Comment by timrae** [comment]
I had also forgotten to push to the release branch after cherry-picking from master and releasing though.

**Comment by mikehardy** [comment]
Great - my local git looks like I'd expect now - :+1: 

Found the problem, OkHTTP is very picky at the moment on version compatibility

v3.12.x is needed until API19 (they dropped support for Android 4.x after this branch)
v3.13 until v4 is needed until API21 (they dropped support for java7 after this branch)

So we need v3.12.x until we move APi above 19

Fixes made and distributed and incorporated here

Lessons learned: we're effectively stable again, so I'll start running [my full-emulator-fleet tests](https://github.com/ankidroid/Anki-Android/blob/master/tools/quality-check/start_all_emulators.sh) prior to any changes going forward and I'll try things on the 2.9 branch too before requesting cherry-picks

And in fact, I've run this on my full fleet - master first (clean, it passed) then this branch (passed), and then on release-2.9 rebased to this branch (it passed).

so TL:DR; @timrae - much better - actually ready to go, fixes us back down to API15, and pre-tested on release-2.9

**Comment by timrae** [comment]
Thanks Mike, I hope you are actually doing some holidaying as well lol

**Comment by mikehardy** [comment]
I am definitely, this is a big-city interlude before one more out-of-touch part of the trip, but my startup pilot was successful (yay) and we're launching 4 weeks into my 6 week "vacation", so...no avoiding everything. And I'm speaking reasonable Spanish in Bolivia which I credit to AnkiDroid so it feels good to keep pushing it forward. Cheers

**Comment by timrae** [comment]
@mikehardy 
Instead of cherry-picking the commits everytime, which is a huge pain and can be quite confusing due to the commit IDs changing, I have finally updated the instructions on the wiki for our old `gitflow` workflow to our new (simpler) branching model. Please see [take a read](https://github.com/ankidroid/Anki-Android/wiki/Release-procedure#merging-new-changes-from-master-into-a-release-or-hotfix-branch) when you get a chance!

**Comment by timrae** [comment]
p.s. @flerda if you see this, you are a genius, your gitflow integrate script is amazing!
